programs/create-genesis/genesis_create.cpp

```diff
@@ -474,10 +478,192 @@ struct genesis_create::genesis_create_impl final {
+                    }
+                } else {
+                    if (l == 1) {
+                        // direct votes at this level
+                        const int max_votes = 30;
+                        const int16_t pct = config::percent_100 / max_votes;
+                        const auto& votes = _visitor.witness_votes[acc];
+                        auto& bps = agents_by_level[0];
+                        auto part = a.balance / max_votes;
+                        int n = 0;
+                        for (const auto& v: votes) {
+                            bool can_vote = bps.count(v) > 0;
+                            if (can_vote) {
+                                n++;
+                                bool last = n == max_votes;
+                                grant(a, bps[v], last ? a.balance : part, pct + (last ? config::percent_100 % max_votes : 0));
+                            } else {
+                                wlog("Skipping ${a} vote for ${w} (not BP)", ("a",names[a.name])("w",_accs_map[v]));
+                            }
+                        }
+                        EOS_ASSERT(n <= _info.params.stake.max_proxies[0], genesis_exception,
+                            "configured max_proxies for level 1 = ${m} is less then actual number votes imported ${a}",
+                            ("m",_info.params.stake.max_proxies[0])("a",n));
+                    } else {
+                        elog("No proxy for level ${l} (${a})", ("l",l)("a",names[a.name]));
+                    }
+                }
+            }
+        }
+        db.start_section(config::system_account_name, N(stake.grant), "grant_struct", grants.size());
+        for (const auto& g: grants) {
+            db.emplace<stake_grant_object>([&](auto& o) {
+                o.token_code = sys_sym.to_symbol_code(),
+                o.grantor_name = g.from,
+                o.agent_name = g.to,
+                o.pct = g.pct,
+                o.share = g.granted;
+                o.granted = g.granted;
+                o.break_fee = 0;
+                o.break_min_own_staked = 0;
+            });
+        }
+
+        db.start_section(config::system_account_name, N(stake.agent), "agent_struct", _visitor.vests.size());
+        for (const auto& abl: agents_by_level) {
+            for (auto& ag: abl) {
+                auto acc = ag.first;
+                auto& x = ag.second;
+                db.emplace<stake_agent_object>([&](auto& a) {
+                    a.token_code = sys_sym.to_symbol_code();
+                    a.account = x.name;
+                    a.proxy_level = x.level;
+                    a.votes = x.level ? 0 : x.balance;
+                    a.last_proxied_update = _conf.initial_timestamp;
+                    a.balance = x.balance;
+                    a.proxied = x.proxied;
+                    a.own_share = x.own_share;
+                    a.shares_sum = x.shares_sum;
+                    a.fee = 0;
```

Comment:
producers should have a 100% fee at the beginning

Reply:
thx